sdk/resourcemanager/azure-resourcemanager-network/src/main/java/com/azure/resourcemanager/network/implementation/WebApplicationFirewallPolicyImpl.java

```diff
@@ -0,0 +1,241 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.resourcemanager.network.implementation;
+
+import com.azure.core.util.CoreUtils;
+import com.azure.resourcemanager.network.NetworkManager;
+import com.azure.resourcemanager.network.fluent.models.WebApplicationFirewallPolicyInner;
+import com.azure.resourcemanager.network.models.ApplicationGateway;
+import com.azure.resourcemanager.network.models.ManagedRuleSet;
+import com.azure.resourcemanager.network.models.ManagedRulesDefinition;
+import com.azure.resourcemanager.network.models.PolicySettings;
+import com.azure.resourcemanager.network.models.WebApplicationFirewallEnabledState;
+import com.azure.resourcemanager.network.models.WebApplicationFirewallMode;
+import com.azure.resourcemanager.network.models.WebApplicationFirewallPolicy;
+import com.azure.resourcemanager.resources.fluentcore.arm.models.implementation.GroupableResourceImpl;
+import com.azure.resourcemanager.resources.fluentcore.utils.ResourceManagerUtils;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Implementation of the {@link WebApplicationFirewallPolicy} interface.
+ */
+public class WebApplicationFirewallPolicyImpl extends GroupableResourceImpl<
+    WebApplicationFirewallPolicy, WebApplicationFirewallPolicyInner, WebApplicationFirewallPolicyImpl, NetworkManager>
+    implements WebApplicationFirewallPolicy,
+    WebApplicationFirewallPolicy.Definition,
+    WebApplicationFirewallPolicy.Update,
+    WebApplicationFirewallPolicy.UpdateStages.WithRequestBodyOrUpdate {
+    private static final String BOT_DETECTION_RULE_SET_TYPE = "Microsoft_BotManagerRuleSet";
+    private static final String BOT_DETECTION_RULE_SET_VERSION_DEFAULT = "0.1";
+    private static final String DEFAULT_MANAGED_RULE_SET = "OWASP";
+    private static final String DEFAULT_MANAGED_RULE_SET_VERSION = "3.2";
```

Comment:
Wondering whether it can be set as only major version e.g. "3.x"

Reply:
Seems not possible.